### PR TITLE
api/consensus/validators: Expose validator signed blocks info

### PR DESCRIPTION
**Pull request by ptrus**
https://app.clickup.com/t/8695px3eh

### CODE REVIEW

File: storage/client/client.go [hunk 1]
```diff
@@ -1303,6 +1303,71 @@ func (c *StorageClient) Validators(ctx context.Context, p apiTypes.GetConsensusV
 		vs.Validators = append(vs.Validators, v)
 	}
 
+	// Validators uptime. TODO: This could be heavily cached or periodically computed in a materialized view.
+	rows, err := c.db.Query(ctx, queries.ValidatorsUptime, address)
```
Comment (ptrus) [inline storage/client/client.go:1307]: TODO: test the validators uptime query performance on a live mainnet DB, to decide if it needs to be moved into a view.

**Review by Andrew7234 — COMMENTED** [review]
Thanks, this impl looks pretty good. Mostly left minor comments; the only major one is about supporting the validator dashboard mocks, which call for information on all validators over the past 24hrs.

File: storage/client/client.go [hunk 2]
```diff
@@ -1303,6 +1303,33 @@ func (c *StorageClient) Validators(ctx context.Context, p apiTypes.GetConsensusV
 		vs.Validators = append(vs.Validators, v)
 	}
 
+	// When querying for a single validator, include the detailed block sign data for last 100 blocks.
```
[the remark this answers is not in the document]
Comment (ptrus) [reply storage/client/client.go:1306]: > Would you mind checking whether it's feasible to return this for all validators

We only need detailed data for the last 100 blocks in the validator details page, so i think we should keep this as it is.

> The validator dashboard calls for an uptime indicator (the green/red uptime vertical bars) for each of the validators in the list, so it'd be nice to package the last 100 blocks info for each validator there

The "uptime bars" and the "% updimte" is for last 24 hours. It is indeed related, but as you mentioned, it should be precomputed on the backend. @jberci is working on that, and we talked a bit about how to do it.

File: storage/client/queries/queries.go [hunk 3]
```diff
@@ -320,6 +320,18 @@ const (
...
+		SELECT
+			height,
+			COALESCE($1 = ANY(signer_entity_ids), FALSE)
+		FROM (
+			SELECT height, signer_entity_ids
+			FROM chain.blocks
+			ORDER BY height DESC
+			LIMIT 100
+		) AS subquery
+		ORDER BY height ASC`
```
[the remark this answers is not in the document]
Comment (ptrus) [reply storage/client/queries/queries.go:333]: Will ask forntend if they care about the order here.

File: storage/client/client.go [hunk 4]
```diff
@@ -1303,6 +1303,33 @@ func (c *StorageClient) Validators(ctx context.Context, p apiTypes.GetConsensusV
...
+				&height,
+				&signed,
+			); err != nil {
+				return nil, wrapError(err)
+			}
+
+			if vs.Validators[0].SignedBlocks == nil {
+				vs.Validators[0].SignedBlocks = new([]ValidatorSignedBlock)
+			}
+			*vs.Validators[0].SignedBlocks = append(*vs.Validators[0].SignedBlocks, ValidatorSignedBlock{
```
[the remark this answers is not in the document]
Comment (ptrus) [reply storage/client/client.go:1326]: Updated.

File: storage/client/queries/queries.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (ptrus) [reply storage/client/queries/queries.go:333]: Updated to return in DESC order and simplified the query.

**Review by Andrew7234 — APPROVED** [review]

**Pull request merged**